File: java/05-04-2024/morning/hospitalJar/src/Pati/Patient.java
package Pati;


public class Patient {
      int patientId;
      protected String patientName;
      protected int bed;
      protected int noOfDays;
      static int count=100;

      public Patient(){
        patientId = ++count;
        bed = 1;
      }

     public void setPatientName(String name){
        patientName = name;
     }

     public String getPatientName(){
        return patientName;
     }

     public void setBedType(int b){
         bed = b;
        }

     public int getBedType(){
        return bed;
     }

     public void setNoOfDays(int days){
        noOfDays = days;
     }

     public int getNoOfDays(){
        return noOfDays;
     }

     public double getPricePerDay(){
        double price = 0;
        switch(bed){
            case 3 :price = 500;
                            break;
            case 2 : price = 350;
                            break;
            default : price = 200;
        }
        return price;
     }

     public double getBillAmount(){
        double amount = noOfDays * getPricePerDay();
        return amount;
     }  
}

File: dotnet/assignments/assignment2/Patient.cs

class Patient
{
    public string patientName{get; set; }

    private static long count = 10000;
    public long patientId = ++count;

    

    public Bed bedType{get; set; }

    public int noOfDays{ get; set; }

    public double getPricePerDay(){
        double price=0;
        switch (bedType){
            case Bed.Executive : price = 500;
                                break;
            case Bed.Special : price = 350;
                                break;
            default : price = 200;
                    break;
        }
        return price;
    }

    public virtual double getBillAmount(){
        return noOfDays * getPricePerDay();
    }
}